Improve ITIL timeline API endpoints

cconard96 opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Needed for #14916 

New:
- Changes timeline item handling to use schemas like every other type.
- Expands the fields for the ITIL type schemas

Fixes:
- Improve RSQL filter handling to skip empty filters
- Remove 'Log" as a timeline item type (Makes more sense to globally handle history)
- Fix issue where each RSQL filter processed would overwrite the array of filters instead of appending (only the last filter would be saved)
- Improve reliability of `RoutePath::getRoutePathWithParameters` by ignoring any array or object parameters passed. These parameters are never valid for the URL so it makes no sense to try and replace placeholders in the URL with their values.

cedric-anne commented on src/Api/HL/Controller/ITILController.php:
```diff
@@ -390,37 +503,93 @@ public function deleteItem(Request $request): Response
+            }
+
+            while ($check < $max_checks && $more_results) {
+                $subitem_results = Search::searchBySchema($schema, [
```
There is a `start` parameter missing here. Each iteration will search for same items.

Also, I wonder if we should limit result. IMHO, it is unlikely to have an ITIL item that have more than 1000 subitems of the same type, and in such case, we cannot even be sure that it will not fail before reaching this limit.

cconard96 commented on src/Api/HL/Controller/ITILController.php:
```diff
@@ -147,21 +174,107 @@ public static function getRawKnownSchemas(): array
                     'format' => Doc\Schema::FORMAT_INTEGER_INT64,
                     'x-readonly' => true,
                 ],
+                'itemtype' => ['type' => Doc\Schema::TYPE_STRING,],
+                'items_id' => ['type' => Doc\Schema::TYPE_INTEGER, 'format' => Doc\Schema::FORMAT_INTEGER_INT64],
```
The x-readonly is not correct for Solution.
From a user point of view, the itemtype and items_id cannot be changed since they are linked to the API path (/Ticket/{id}/...). From the API point though, the fields have to be writable when the item is being created as the controller manually sets these fields while readonly fields are filtered out in at the Search level.

cedric-anne reviewed (changes requested):
Tests are failing.

cedric-anne commented:
@cconard96 

Please solve conflicts.

cconard96 commented:
Rebased. Last change added was to add the missing `documents_id` field for `Document_Item` and also added the `filepath` property (mainly for ease of use with the Webhooks feature).
